pkg/client/events/replay_client.go

```diff
@@ -198,27 +203,13 @@ func (rClient *replayClient[T, R]) goPublishEvents(ctx context.Context) {
 // replayObsCache replay observable.
 func (rClient *replayClient[T, R]) retryPublishEventsFactory(ctx context.Context) func() chan error {
 	return func() chan error {
-		logger := polylog.Ctx(ctx)
-		logger.Debug().
-			Msg("retryPublishEventsFactory: creating new events bytes observable")
 		errCh := make(chan error, 1)
 		eventsBzObs, err := rClient.eventsClient.EventsBytes(ctx, rClient.queryString)
 		if err != nil {
 			errCh <- err
 			return errCh
 		}
 
-		// Subscribe to the eventBzObs and block until the channel closes.
-		// Then pass this as an error to force the retry.OnError to resubscribe.
-		go func() {
-			eventBzObsSub := eventsBzObs.Subscribe(ctx)
-			for range eventBzObsSub.Ch() {
-				// Do nothing, just wait for the channel to close.
-				continue
-			}
-			errCh <- fmt.Errorf("EventsQueryClient closed")
-		}()
-
 		// NB: must cast back to generic observable type to use with Map.
 		eventsBz := observable.Observable[either.Either[[]byte]](eventsBzObs)
```

Comment:
NIT: Can we rename this to `eventsBzObs`?

It'll match the comment you added below as well.